R-packages/covidcast/R/covidcast.R

```diff
@@ -529,58 +532,59 @@ covidcast_days <- function(data_source, signal, start_day, end_day, geo_type,
   for (i in seq(ndays + 1)) {
     query_day <- start_day + i - 1
     day_str <- date_to_string(query_day)
-    dat[[i]] <- covidcast(data_source = data_source,
-                          signal = signal,
-                          time_type = "day",
-                          geo_type = geo_type,
-                          time_values = day_str,
-                          geo_value = geo_value,
-                          as_of = as_of,
-                          issues = issues,
-                          lag = lag)
+    res <- covidcast(data_source = data_source,
+                     signal = signal,
+                     time_type = "day",
+                     geo_type = geo_type,
+                     time_values = day_str,
+                     geo_value = geo_value,
+                     as_of = as_of,
+                     issues = issues,
+                     lag = lag)
+
+    if (is.null(res)) {
+      warn(paste0("Fetching ", signal, " from ", data_source, " for ",
+                  query_day, " in geography '", geo_value, "': no results"),
+           data_source = data_source,
+           signal = signal,
+           day = query_day,
+           geo_value = geo_value,
+           class = "covidcast_fetch_failed")
+
+      next
+    }
+
+    dat[[i]] <- res
+
     summary <- sprintf(
-      "Fetched day %s: %s, %s, num_entries = %s",
+      "Fetched day %s: num_entries = %s",
       query_day,
-      dat[[i]]$result,
-      dat[[i]]$message,
-      nrow(dat[[i]]$epidata)
+      nrow(dat[[i]])
     )
     if (length(summary) != 0) {
       message(summary)
     }
-    if (dat[[i]]$message == "success") {
-      returned_geo_values <- dat[[i]]$epidata$geo_value
-      if (!identical("*", geo_value)) {
-        missed_geos <- setdiff(tolower(geo_value),
-                               tolower(returned_geo_values))
-        if (length(missed_geos) > 0) {
-          missed_geos_str <- paste0(missed_geos, collapse = ", ")
-          warn(sprintf("Data not fetched for some geographies on %s: %s",
-                         query_day, missed_geos_str),
-               data_source = data_source,
-               signal = signal,
-               day = query_day,
-               geo_value = geo_value,
-               api_msg = dat[[i]]$message,
-               class = "covidcast_missing_geo_values"
-               )
-        }
+
+    if (nrow(dat[[i]]) > 0 && !identical("*", geo_value)) {
```

Comment:
Why are we using `identical()` and not just `==`?

Reply:
If `geo_value` is a vector, `"*" == geo_value` will be a vector of logicals instead of one boolean value.